Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,258 @@
+            }
+
+            if (name == "")
+            {
+                return NotAvailable;
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var biggestLossEverNoComma = RemoveComma($"{biggestLossEver:C0}").Replace("(", "-").Replace(")", "");
+            return $"{name} lost the most money. {biggestLossEverNoComma}.";
+        }
+
+        private decimal GetBiggestLossForPerson(string[] peopleBalances)
+        {
+            var biggestLoss = decimal.MaxValue;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                for (var j = i + 1; j < peopleBalances.Length; j++)
+                {
+                    var isBeforeBalanceValid = decimal.TryParse(peopleBalances[i], out var beginAmount);
+                    var isAfterBalanceValid = decimal.TryParse(peopleBalances[j], out var endAmount);
+
+                    if (isBeforeBalanceValid && isAfterBalanceValid)
+                    {
+                        biggestLoss = GetBiggestLoss(biggestLoss, beginAmount, endAmount);
+                    }
+                }
+            }
+
+            return biggestLoss;
+        }
+
+        private static decimal GetBiggestLoss(decimal biggestLoss, decimal beginAmount, decimal endAmount)
+        {
+            var loss = endAmount - beginAmount;
+            var bigLoss = biggestLoss;
+
+            if (loss < bigLoss)
+            {
+                bigLoss = loss;
+            }
+
+            return bigLoss;
+        }
+
+        /// <summary>
+        /// Return name and current money of the richest person.
+        /// </summary>
+        public string FindRichestPerson(string[] peopleAndBalances)
+        {
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return NotAvailable;
+            }
+
+            var maxBalance = 0m;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid)
+                {
+                    SetNames(ref name, ref maxBalance, peopleBalances, balance);
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word1 = name.ToString().Contains(", ") ? "are" : "is";
+            var word2 = name.ToString().Contains(", ") ? "people" : "person";
+            return $"{FixPlural(name.ToString())} {word1} the richest {word2}. {maxBalanceNoComma}.";
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public string FindMostPoorPerson(string[] peopleAndBalances)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'FindMostPoorPerson' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)